plugins/jobs/src/js/components/form/helpers/MetronomeJobValidators.ts

```diff
@@ -219,29 +276,165 @@ export const MetronomeSpecValidators: MetronomeValidators = {
+        message: numberErrorMessage
+      });
+    }
+    if (mem != undefined && typeof mem !== "number") {
+      errors.push({
+        path: ["job", "run", "mem"],
+        message: numberErrorMessage
+      });
+    }
+    if (disk != undefined && typeof disk !== "number") {
+      errors.push({
+        path: ["job", "run", "disk"],
+        message: numberErrorMessage
+      });
+    }
+    return errors;
+  },
+
+  checkTypesOfDockerProps(formData: JobOutput) {
+    const docker = findNestedPropertyInObject(formData, "job.run.docker");
+    const errors: FormError[] = [];
+
+    if (docker == undefined) {
+      return errors;
+    }
+
+    const image = findNestedPropertyInObject(formData, "job.run.docker.image");
+    const forcePullImage = findNestedPropertyInObject(
+      formData,
+      "job.run.docker.forcePullImage"
+    );
+    const privileged = findNestedPropertyInObject(
+      formData,
+      "job.run.docker.privileged"
+    );
+    // Docker parameters are checked separately.
+
+    if (image != undefined && typeof image !== "string") {
+      errors.push({
+        path: ["job", "run", "docker", "image"],
+        message: stringErrorMessage
+      });
+    }
+    if (forcePullImage != undefined && typeof forcePullImage !== "boolean") {
+      errors.push({
+        path: ["job", "run", "docker", "forcePullImage"],
+        message: booleanErrorMessage
+      });
+    }
+    if (privileged != undefined && typeof privileged !== "boolean") {
+      errors.push({
+        path: ["job", "run", "docker", "privileged"],
+        message: booleanErrorMessage
+      });
+    }
+    return errors;
+  },
+
+  checkTypesOfUcrProps(formData: JobOutput) {
+    const ucr = findNestedPropertyInObject(formData, "job.run.ucr");
+    const errors: FormError[] = [];
+
+    if (ucr == undefined) {
+      return errors;
+    }
+
+    const image = findNestedPropertyInObject(formData, "job.run.ucr.image");
+    const privileged = findNestedPropertyInObject(
+      formData,
+      "job.run.ucr.privileged"
+    );
+    const id = findNestedPropertyInObject(formData, "job.run.ucr.image.id");
+    const kind = findNestedPropertyInObject(formData, "job.run.ucr.image.kind");
+    const forcePull = findNestedPropertyInObject(
+      formData,
+      "job.run.ucr.image.forcePull"
+    );
+
+    if (image != undefined && typeof image !== "object") {
+      errors.push({
+        path: ["job", "run", "ucr", "image"],
+        message: objectErrorMessage
+      });
+    }
+    if (privileged != undefined && typeof privileged !== "boolean") {
+      errors.push({
+        path: ["job", "run", "ucr", "privileged"],
+        message: booleanErrorMessage
+      });
+    }
+    if (id != undefined && typeof id !== "string") {
+      errors.push({
+        path: ["job", "run", "ucr", "image", "id"],
+        message: stringErrorMessage
+      });
+    }
+    if (
+      kind != undefined &&
+      (kind !== UcrImageKind.Docker || kind !== UcrImageKind.Appc)
```

Comment:
This is always going to be true. since kind can't be `Appc` and `Docker` at the same time. Based on the message I guess we want to have `&&` instead of the `||`

Reply:
Ah you are so right! Thank you!